pylint/utils.py

```diff
@@ -730,32 +732,93 @@ def register_messages(self, checker):
             chkid = msg.msgid[1:3]
             self._messages[msg.symbol] = msg
-            self._register_alternative_name(msg, msg.msgid)
+            self._register_alternative_name(msg, msg.msgid, msg.symbol)
             for old_id, old_symbol in msg.old_names:
-                self._register_alternative_name(msg, old_id)
-                self._register_alternative_name(msg, old_symbol)
+                self._register_alternative_name(msg, old_id, old_symbol)
             self._msgs_by_category[msg.msgid[0]].append(msg.msgid)
 
-    def _register_alternative_name(self, msg, name):
+    def _register_alternative_name(self, msg, msgid, symbol):
         """helper for register_message()"""
-        if name in self._messages and self._messages[name] != msg:
-            raise InvalidMessageError(
-                'Message symbol %r is already defined' % name)
-        if name in self._alternative_names and self._alternative_names[name] != msg:
-            raise InvalidMessageError(
-                'Message %s %r is already defined' % (
-                    'id' if len(name) == 5 and name[0] in MSG_TYPES else 'alternate name',
-                    name))
-        self._alternative_names[name] = msg
+        self._check_id_and_symbol_consistency(msgid, symbol)
+        self._alternative_names[msgid] = msg
+        self._alternative_names[symbol] = msg
+
+    def _check_symbol(self, msgid, symbol):
+        """ Check that a symbol is not already used. """
+        if symbol in self._messages:
+            other_msgid = self._messages[symbol].msgid
+            self._raise_duplicate_msg_id(symbol, msgid, other_msgid)
+        if symbol in self._alternative_names:
+            msg = self._alternative_names[symbol]
+            if msg.symbol == symbol:
+                other_msgid = msg.msgid
+            else:
+                for old_msgid, old_symbol in msg.old_names:
+                    if old_symbol == symbol:
+                        other_msgid = old_msgid
+                        break
+            if msgid != other_msgid:
+                self._raise_duplicate_msg_id(symbol, msgid, other_msgid)
+
+    def _check_msgid(self, msgid, symbol):
+        for m_symbol in self._messages:
+            msg = self._messages[m_symbol]
+            if msg.msgid == msgid:
+                self._raise_duplicate_symbol(msgid, symbol, msg.symbol)
+
+    def _check_message(self, msg, check_id=None, is_old_name=False):
+        """A new messages is correct according to a dict of messages. """
+        msgid, symbol, old_names = msg.msgid, msg.symbol, msg.old_names
+        self._check_id_and_symbol_consistency(msgid, symbol)
+        self._check_symbol(msgid, symbol)
+        self._check_msgid(msgid, symbol)
+        for old_name in old_names:
+            self._check_symbol(msgid, old_name[1])
+        if check_id is not None and check_id != msgid[1:3]:
+            error_msg = "Inconsistent checker part in message id %r" % msgid
```

Comment:
The master branch runs completely on Python 3, so feel free to use f-strings here instead:
`f'Inconsistent .. in message id {msgid}`.

Reply:
Scratch that suggestion, we still need to support Python 3.4 and 3.5, so no f-strings for now.